### PR TITLE
UI: Remove support for numeric multiplier on Box padding

### PR DESCRIPTION
**Pull request by aduth**
## What?

Updates `@wordpress/ui`'s `Box` component to remove support for passing a number as a padding value, requiring consumers to use one of the available token values.

Extracted from #74001 (see https://github.com/WordPress/gutenberg/pull/74001#discussion_r2620316479).

## Why?

See #73852 for context:

> Since this component is a foundational component of the design system, it should aim to build on design tokens that form the basis of expressing spacing in a systems-based manner.

## Testing Instructions

None of the current stories or tests provide coverage for this, likely because it was always considered an edge case, now completely unsupported.

You can still verify there is no impact to the rendering behavior of existing Storybook stories:

1. `npm run storybook:start`
2. Go to http://localhost:50240/?path=/docs/design-system-components-box--docs
3. Verify no regression in appearance of `Box`

### CODE REVIEW

**Review by jameskoster — APPROVED** [review]

File: packages/ui/src/box/types.ts
Comment (mirka) [inline packages/ui/src/box/types.ts]: By the way, was there any reason why the `target` type is not a union of valid strings?

File: packages/ui/src/box/box.tsx [hunk 1]
```diff
@@ -29,21 +29,19 @@ const capitalize = ( str: string ): string =>
 
 /**
  * Converts a size value to a CSS design token property reference (with
```
Comment (mirka) [inline packages/ui/src/box/box.tsx]: ?

```suggestion
 * Converts a size token name to a CSS design token property reference (with
```

**Review by mirka — APPROVED** [review]

File: packages/ui/src/box/types.ts
Comment (aduth) [reply packages/ui/src/box/types.ts]: I seem to recall some earlier consideration for using custom, extended targets, but now that you mention it, I'm not sure that's something we can or want to support.

Comment (aduth) [reply packages/ui/src/box/types.ts]: > I seem to recall some earlier consideration for using custom, extended targets, but now that you mention it, I'm not sure that's something we can or want to support.

More context: https://github.com/WordPress/gutenberg/pull/72984#issuecomment-3491643955

Specifically this comment:

> Part of the expectation here is that in how some components would use Box, they may introduce their own family of tokens, like a `Card` component using `Box` and having its own card-specific spacing tokens

Realistically, the way we're implementing tokens would mean that having tokens outside what exists today would still be something we'd want to include as a peer option of the existing `surface`, `interactive` options, and as such "a union of valid strings" can follow from that.

tl;dr: Yes, we should use a union 😄

File: packages/ui/src/box/box.tsx (hunk 1 above)
Comment (aduth) [reply packages/ui/src/box/box.tsx]: Yeah, that's clearer now that we're only accepting one value type 👍

File: packages/ui/src/box/types.ts
Comment (aduth) [reply packages/ui/src/box/types.ts]: Updated in 6cf0780d81f. Separately I'd like to explore a way to better automate these types out of our existing token files.
